src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -14,6 +15,7 @@ namespace Microsoft.PowerFx.Core.Public.Types
     /// Formula Types are a class hiearchy.
     /// </summary>
     [DebuggerDisplay("{_type}")]
+    [ImmutableObject(true)]
```

Comment:
>[ImmutableObject(true)] [](http://example.com/codeflow?start=4&length=23)

per the test, we could mark "safe" classes as immutable (either via the Sys.ComponentModel attribute or creating our own). 
This just tells the test it's safe; it doesn't actually enforce it (although maybe there's a followup way to enforce classes marked as immutable are actually immutable)

Reply:
We also have a potential hole to ensure that the *derived* classes are immutable.